router/core/header_rule_engine_test.go

```diff
@@ -167,3 +169,69 @@ func TestInvalidRegex(t *testing.T) {
 	})
 	assert.Error(t, err)
 }
+
+func TestSubgraphPropagateHeaderRule(t *testing.T) {
+	ht, err := NewHeaderTransformer(config.HeaderRules{
+		Subgraphs: map[string]config.GlobalHeaderRule{
+			"subgraph-1": {
+				Request: []config.RequestHeaderRule{
+					{
+						Operation: "propagate",
+						Named:     "X-Test-Subgraph-1",
+						Default:   "Test-Value-1",
+					},
+				},
+			},
+			"subgraph-2": {
+				Request: []config.RequestHeaderRule{
+					{
+						Operation: "propagate",
+						Named:     "X-Test-Subgraph-2",
+						Default:   "Test-Value-2",
```

Comment:
Please ensure we have test cases that cover the default and propagation case. Currently, we have two equal test cases.

Reply:
In order to resolve this, you would need to remove the default value of one of these subgraphs.